packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -214,6 +215,40 @@ export class CacheService {
     }
   }
 
+  /**
+   * Sets multiple values in the cache, each associated with its respective key.
+   * If the shared cache is enabled and an error occurs while setting in it,
+   * the internal cache is used as a fallback.
+   * @param {Record<string, any>} entries - An object containing key-value pairs to cache.
+   * @param {string} callingMethod - The name of the method calling the cache.
+   * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
+   * @param {boolean} shared - Whether to use the shared cache (optional, default: false).
+   */
+  public multiSet(
+    entries: Record<string, any>,
+    callingMethod: string,
+    requestIdPrefix?: string,
+    shared: boolean = true,
+  ): void {
+    if (shared && this.isSharedCacheEnabled) {
+      try {
+        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.REDIS, CacheService.methods.MSET).inc(1);
+        this.sharedCache.multiSet(entries, callingMethod, requestIdPrefix);
+      } catch (error) {
+        const redisError = new RedisCacheError(error);
+        this.logger.error(
+          `${requestIdPrefix} Error occurred while setting the cache to Redis. Fallback to internal cache. Error is: ${redisError.fullError}`,
+        );
+        // Fallback to internal cache
+        this.internalCache.multiSet(entries, callingMethod, requestIdPrefix);
```

Comment:
Missing metric here in the fall back case

Reply:
✅ Done